pandas-stubs/core/indexes/multi.pyi

```diff
@@ -27,31 +29,42 @@ from pandas._typing import (
 class MultiIndex(Index[Any]):
     def __new__(
         cls,
-        levels=...,
-        codes=...,
-        sortorder=...,
+        levels: SequenceNotStr[SequenceNotStr[Hashable] | AnyArrayLike] = ...,
```

Comment:
```suggestion
        levels: Sequence[SequenceNotStr[Hashable]] | AnyArrayLike] = ...,
```
The outer sequence doesn't need to be `NotStr` because the inner one is already not a string.